packages/core/echo/echo-db/src/client/doc-handle-client.test.ts

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { expect } from 'chai';
+
+import { generateAutomergeUrl, parseAutomergeUrl, Repo } from '@dxos/automerge/automerge-repo';
+import { DocSyncState } from '@dxos/echo-pipeline';
+import { describe, test } from '@dxos/test';
+
+import { DocHandleClient } from './doc-handle-client';
+
+describe('DocHandleClient', () => {
+  test('get update from handle', async () => {
+    const text = 'Hello World!';
+    const { documentId } = parseAutomergeUrl(generateAutomergeUrl());
+    const docHandle = new DocHandleClient<{ text: string }>(documentId);
+    docHandle.change((doc: { text: string }) => {
+      doc.text = text;
+    });
+
+    const repoToSync = new Repo({ network: [] });
+    const handleToSync = repoToSync.find(documentId);
+    const syncState = new DocSyncState(handleToSync);
```

Comment:
i don't see why we need to create a new object here (see note above -- DocSyncState probably doesn't need to be a class).